src/main/java/fr/sncf/osrd/DebugViewer.java

```diff
@@ -263,9 +264,13 @@ private void updateTrain(TrainData trainData) {
         var nextMove = trainData.nextMove;
 
         var lastUpdate = nextMove.findLastSpeedUpdate(currentTime);
+        if (lastUpdate == null)
+            return;
         var pathHeadPosition = lastUpdate.interpolatePosition(currentTime);
         var pathTailPosition = Double.max(0, pathHeadPosition - trainData.schedule.rollingStock.length);
         var headLocation = trainData.schedule.findLocation(pathHeadPosition);
+        if (headLocation == null)
+            return;
```

Comment:
This seems like an error case, do we want to ignore it? Or is it an usual state?

Reply:
This case occurs when the train finishes its phase with a non-zero speed.